javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/SafeLogger.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Debug logging that is performed under class file transform needs to use this class, because
+ * gradle deadlocks sporadically under the following sequence:
+ * <li>Gradle triggers a class to load while it is holding a lock
+ * <li>Class file transform occurs (under this lock) and the agent writes to System.out
+ * <li>(Because gradle hijacks System.out), gradle is called from inside of the class file transform
+ * <li>Gradle tries to grab a different lock during it's implementation of System.out
+ */
+public class SafeLogger {
```

Comment:
```suggestion
public class GradleSafeLogger {
```

maybe

Reply:
I like this name, it does look a little weird in the places where it's used tho, related to @iNikem's point about any mention of gradle being weird, will sleep on it 😄